pytorch_lightning/strategies/bagua.py

```diff
@@ -165,11 +160,12 @@ def configure_ddp(self) -> None:
         self._model = self._setup_model(model)
 
         # start the background communication for async algorithm
+        assert self.lightning_module is not None
         assert self.lightning_module.trainer is not None
         if self.lightning_module.trainer.training and self._bagua_algorithm == "async":
             self.model.bagua_algorithm.resume(self.model)  # type: ignore
 
-    def _setup_model(self, model: Module) -> BaguaDistributedDataParallel:
+    def _setup_model(self, model: Module) -> "BaguaDistributedDataParallel":
```

Comment:
Was it wrong before? Just curious!

Reply:
With the old style it would lead to errors if the import at the top could not be resolved. This shouldn't matter for anything but typing :)

Reply:
Yes it was wrong, because `BaguaDistributedDataParallel` was set to `None` when `_IS_BAGUA_AVAILABLE = False ` (see imports above). This is a typing issue and shows up as a big warning in the IDE ^^